arbor/include/arbor/recipe.hpp

```diff
@@ -54,8 +54,8 @@ struct cell_connection_base {
 
     cell_connection_base(L src, cell_local_label_type dst, float w, const U::quantity& d):
         source(std::move(src)), target(std::move(dst)), weight(w), delay(d.value_as(U::ms)) {
-        if (std::isnan(weight)) throw std::out_of_range("Connection weight must be finite.");
-        if (std::isnan(delay) || delay < 0)  throw std::out_of_range("Connection delay must be non-negative and infinite in units of [ms].");
+        if (std::isnan(weight)) throw std::domain_error("Connection weight must be finite.");
+        if (std::isnan(delay) || delay <= 0)  throw std::domain_error("Connection delay must be positive, finite, and given in units of [ms].");
     }
```

Comment:
We have quite specific exceptions such as `arb::bad_connection_source_gid` defined in `<arbexcept.hpp>`; it would be consistent to define some exceptions to throw here that derive from `arb::arbor_exception`.

Reply:
I am a bit torn here, `std::domain_error` fits the basic checks for `isnan` etc well and has the advantage that it plays nice with pybind11 (it'll automatically convert to `ValueError`).